src/pluginLoader.js

```diff
@@ -0,0 +1,660 @@
+    });
+
+    var BootstrapPlugin = new BootstrapPluginClass();
+
+    /**
+     * This plugin's name and angular module
+     */
+    BootstrapPlugin.pluginName = "bootstrap-plugin";
+    /**
+     * This plugins logger instance
+     */
+    var log = Logger.get(BootstrapPlugin.pluginName);
+
+    var _module = angular.module(BootstrapPlugin.pluginName, []);
+    _module.config(["$locationProvider", function ($locationProvider) {
+      $locationProvider.html5Mode(true);
+    }]);
+
+    _module.run(['documentBase', function (documentBase) {
+      log.debug("loaded");
+    }]);
+
+    BootstrapPlugin.documentBase = function() {
+      var base = $('head').find('base');
+      var answer = '/'
+      if (base && base.length > 0) {
+        answer = base.attr('href');
+      } else {
+        log.warn("Document is missing a 'base' tag, defaulting to '/'");
+      }
+      return answer;
+    }
+
+    // Holds the document base so plugins can easily
+    // figure out absolute URLs when needed
+    _module.factory('documentBase', function() {
+      return BootstrapPlugin.documentBase();
+    });
+
+    pluginLoader.addModule("ng");
+    pluginLoader.addModule("ngSanitize");
+    pluginLoader.addModule(BootstrapPlugin.pluginName);
+
+    // bootstrap the app
+    $(function () {
+
+      jQuery.uaMatch = function( ua ) {
+        ua = ua.toLowerCase();
+
+        var match = /(chrome)[ \/]([\w.]+)/.exec( ua ) ||
+          /(webkit)[ \/]([\w.]+)/.exec( ua ) ||
+          /(opera)(?:.*version|)[ \/]([\w.]+)/.exec( ua ) ||
+          /(msie) ([\w.]+)/.exec( ua ) ||
+          ua.indexOf("compatible") < 0 && /(mozilla)(?:.*? rv:([\w.]+)|)/.exec( ua ) ||
+          [];
+
+        return {
+          browser: match[ 1 ] || "",
+          version: match[ 2 ] || "0"
+        };
+      };
+
+      // Don't clobber any existing jQuery.browser in case it's different
+      if ( !jQuery.browser ) {
+        var matched = jQuery.uaMatch( navigator.userAgent );
+        var browser = {};
+
+        if ( matched.browser ) {
+          browser[ matched.browser ] = true;
+          browser.version = matched.version;
+        }
+
+        // Chrome is Webkit, but Webkit is also Safari.
+        if ( browser.chrome ) {
+          browser.webkit = true;
+        } else if ( browser.webkit ) {
+          browser.safari = true;
+        }
+
+        jQuery.browser = browser;
+      }
+
+      pluginLoader.loadPlugins(function() {
+
+        if (BootstrapPlugin.injector) {
+          log.debug("Application already bootstrapped");
+          return;
+        }
+
+        var bootstrapEl = pluginLoader.getBootstrapElement();
+        log.debug("Using bootstrap element: ", bootstrapEl);
+
+        BootstrapPlugin._injector = angular.bootstrap(bootstrapEl, pluginLoader.getModules(), {
+          strictDi: false
+        });
+        log.debug("Bootstrapped application");
+      });
+    });
+    return BootstrapPlugin;
+})();
+
+var hawtioPluginLoader = pluginLoader;
```

Comment:
Let's make it explicitly `window.hawtioPluginLoader = pluginLoader` so we're sure it's always a global.